uswds/uswds-site
# USWDS-Site - Changelog: add entry and doc updates for sticky table headers [USWDS#5420]

# Summary
PR #5420 introduced  sticky table headers as a new variant for USWDS tables. This PR adds documentation for the new variant. 

- Create changelog entries for uswds/uswds#5420
- Add documentation for new table variant
- Add documentation for new table settings

> [!IMPORTANT]
> This branch will have build failures until uswds is updated to include the changes in 3.8.0 (particularly, uswds/uswds#5420).

### New variant

| Variant name              | Description | 
| ---------------------------- | :-------------- |
| `.usa-table--sticky-header` |   Allows the table header element to stick to the top of the page as the user scrolls. Preferred for long tables with many rows. This variant is not compatible with the scrollable and stacked table variants. |   

### New settings

| Setting name              | Description | 
| ---------------------------- | :-------------- |
| `$theme-table-background-color` |   Defines the table background color. A value of `default` uses the body background color. |   
| `$theme-table-sticky-top-offset` |  Defines the top offset for sticky headers. |

## Preview link
- [Sticky table header variant](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-documentation-5420/components/table/#sticky-table-headers)
- [Table variant list](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-documentation-5420/components/table/#component-variants-table)
- [Table settings list](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-documentation-5420/components/table/#component-settings-table)
- [Table changelog](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-documentation-5420/components/table/#changelog)
- [Settings page - table](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-documentation-5420/documentation/settings/#tables)
- [Settings page changelog](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-documentation-5420/documentation/settings/#changelog)

## Testing and review
- Confirm the changelog entries make sense and are added to the appropriate pages
- Confirm the settings and variant updates make sense and are added to the appropriate pages
- Confirm grammar, spelling, and plain language for all copy updates
- Confirm that there are no visual regressions in any of the table variants affected by uswds/uswds#5420
- Confirm the sticky table header works as expected in the component preview
